### PR TITLE
feat(build): #1159 excessive privileges

### PR DESCRIPTION
**Pull request by rohaquinlop**
- Solve excessive privileges to temporary files
- Creates a temporary file in the most secure manner possible.

### CODE REVIEW

File: src/cli/main/cli.py [hunk 1]
```diff
@@ -411,11 +411,16 @@ class Config(NamedTuple):
     cache: List[Dict[str, str]]
 
 
+def _get_named_temporary_file_name() -> str:
```
Comment (dsalaza4) [inline src/cli/main/cli.py:414]: @rohaquinlop Sometimes we get an error saying that a temporary directory already exists when we run several instances of Makes at the same time. Can you please extend this function so it waits 1 second and retries if the directory already exists? Let's retry 5 times before failing.

Comment (dsalaza4) [reply src/cli/main/cli.py:414]: Related issue is https://github.com/fluidattacks/makes/issues/1068

Comment (rohaquinlop) [reply src/cli/main/cli.py:414]: Ok, got it

File: src/cli/main/cli.py [hunk 2]
```diff
@@ -411,11 +412,28 @@ class Config(NamedTuple):
...
+    return file_name
+
+
 def _get_config(head: str, attr_paths: str) -> Config:
     CON.out()
     CON.rule("Building project configuration")
     CON.out()
-    out: str = tempfile.mktemp()  # nosec
+
+    attempts = 0
```
Comment (dsalaza4) [inline src/cli/main/cli.py:428]: Let's keep this logic within `_get_named_temporary_file_name `

File: src/cli/main/cli.py [hunk 3]
```diff
@@ -411,11 +412,28 @@ class Config(NamedTuple):
     cache: List[Dict[str, str]]
 
 
+def _get_named_temporary_file_name() -> str:
```
Comment (dsalaza4) [inline src/cli/main/cli.py:415]: Please test this change in universe with the `m . /integrates` job

Comment (rohaquinlop) [reply src/cli/main/cli.py:415]: Works!
image

File: src/cli/main/cli.py [hunk 4]
```diff
@@ -411,11 +412,32 @@ class Config(NamedTuple):
...
+    attempts = 0
+    file_name = ""
+    while attempts < 4 and file_name == "":
+        try:
+            with tempfile.NamedTemporaryFile(delete=True) as file:
+                file_name = file.name
+            sleep(1)
```
Comment (dsalaza4) [inline src/cli/main/cli.py:422]: Unnecessary sleep? I think the second one is enough

File: src/cli/main/cli.py [hunk 5]
```diff
@@ -411,11 +412,32 @@ class Config(NamedTuple):
...
+        except FileExistsError as error:
+            CON.print(f"Failed to create {error.filename}, retrying in 1 second...")
+            attempts += 1
+            sleep(1)
+
+    if file_name == "":
```
Comment (dsalaza4) [inline src/cli/main/cli.py:428]: I don't understand the purpose of this. Maybe we should raise an exception if all 5 attempts are failed?

Comment (rohaquinlop) [reply src/cli/main/cli.py:428]: The loop tries 4 times, this is the fifth and last and if it still fails then will raise the error

File: src/cli/main/cli.py (hunk 4 above)
Comment (rohaquinlop) [reply src/cli/main/cli.py:422]: Removing it

File: src/cli/main/cli.py (hunk 5 above)
Comment (dsalaza4) [reply src/cli/main/cli.py:429]: Handle it with a boolean so you don't have to repeat the logic outside the while cycle?:

```
success = False

while:
  ...
if not sucess:
  raise exception
```

File: src/cli/main/cli.py [hunk 6]
```diff
@@ -411,11 +412,33 @@ class Config(NamedTuple):
...
+                success = True
+        except FileExistsError as error:
+            CON.print(
+                f"Failed to create {error.filename}, retrying in 1 second..."
+            )
+            attempts += 1
+            sleep(1)
+
+    if not success:
+        raise FileExistsError("Temporary file already exists.")
```
Comment (dsalaza4) [inline src/cli/main/cli.py:432]:
```suggestion
        raise FileExistsError("Could not create file after 5 attempts.")
```

**Review by dsalaza4 — APPROVED** [review]
lgtm

**Pull request merged**